### DISCUSSION TITLE
Loopback interface visible

### DISCUSSION THREAD

**Discussion by RogueWarrior65** [RaspAP/raspap-webgui#849] [General]
FYI, the lo (loopback) network interface is visible in the Hotspot page and the DHCP Server page popup menus.

`exec("ip -o link show | awk -F': ' '{print $2}'", $interfaces);`

in includes/dhcp.php should read:

`exec("ip -o link show | awk -F': ' '{print $2}' | grep -v lo", $interfaces);`

In my particular embedded system, I also have two CAN Bus interfaces (can0 and can1) which should be filtered out.
I can accomplish this by using this line instead:

`exec("ip -o link show | awk -F': ' '{print $2}' | grep -v lo | grep -Ev 'can*'", $interfaces);`

I also have to modify includes/networking.php and get_all_interfaces.php

`exec("ls /sys/class/net | grep -v lo", $interfaces);`

to

`exec("ls /sys/class/net | grep -v lo | grep -Ev 'can*'", $interfaces);`

There are probably a few other places where building a dynamic list of available network interfaces is done.  Perhaps there should be one main function to do this.

**Comment by billz** [comment]
A common function to enumerate interfaces is a good suggestion. One of the goals of this project is to make it easy to customize without touching code. To that end, the .json files in /config are used to store various default values. A list of filtered interfaces is a good candidate for addition to `defaults.json`.

**Comment by glaszig** [comment]
since one can basically name interfaces however one wants and have aliases and bridges and whatever, would it not be better to have a filter function or filter regexp config option?

**Reply by billz** [reply]
yes, something along these lines would also work.
